design/specs/foreign-key-constraints.md

```diff
@@ -2,75 +2,169 @@
+- The user selects a table to be referenced by these columns.
+- The user selects a column in the reference table from which the column will match values. By default, this is set to the 'Primary Key' column in the referenced table. When changing this field, the user can select only columns with a UNIQUE or PRIMARY KEY constraint.
 
----
+## Scenario 2: User Edits an Existing Foreign Key Constraint
+
+### Scenario 2a: From the column header menu
+
+#### Steps for 2a
+
+- The user hovers over the foreign key reference link below the column name.
+- Information about the constraints applied to the column is displayed in a tooltip.
+- The user clicks on the foreign key reference link below the column name.
+- Constraint settings open for the column.
+
+### Scenario 2b: From table constraint settings
+
+#### Steps for 2b
+
+- A user wants to edit an existing foreign key constraint to change the selected columns or select another referenced table.
+- The user can open a specific foreign key constraint form by clicking on the foreign key link in the column header.
+- If the modified FK constraint causes existing column values to become invalid, the user will receive a warning and be aware of the potential implications of this change, like loss of data or broken views.
+
+## Scenario 3: User Identifies a Column With a Foreign Key Constraint Applied
+
+### Scenario 3a: The foreign key constraint is set to a single column
+
+#### Steps for 3a
+
+- The user opens a table that contains columns with a foreign key constraint applied.
+- The user identifies the columns in two ways:
+  - The user looks at the column header and sees the foreign key indicator. This indicator shows a key icon and the referenced table and column next to it.
+    - The user can click the indicator to display the foreign key settings for the specific column.
+    - The indicator can be hovered over to display a tooltip with information about the foreign key constraints.
+  - The user looks at the column fields and sees that values are added with different styling, a tag-like element with a colored background.
+
+### Scenario 3b: The foreign key constraint is set to multiple columns
+
+#### Steps for 3b
+
+- The user opens a table that contains columns with a multi-column foreign key constraint applied.
+  - The indicator, in this case, looks similar to the single-column one. However, the user will understand that the foreign key constraints are referencing the same table.
+    - The user can tell that a referenced table is the same because they share the same color. The system could use other UI elements instead of color. Still, the idea is to create a visual distinction that can be easily scanned.
+  - The indicator can be hovered over to see additional information. From here, the user will understand that the constraint references more than one column.
+
+## Scenario 4: User Sees a Preview of the Linked Record in a Column With a Foreign Key Constraint Applied
```

Comment:
We'll have a similar UX challenge (identifying the linked records) for multi-column foreign key constraints, but I don't understand whether this UI will assist with that.

Reply:
I've added a note to add this to a future design issue. For now, we'll assume it's a single-column constraint.